extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/v1/GreenbidsRealTimeDataProcessedAuctionRequestHook.java

```diff
@@ -0,0 +1,219 @@
+                    storage,
+                    gcsBucketName,
+                    onnxModelCacheKeyPrefix,
+                    thresholdsCacheKeyPrefix);
+            onnxModelRunner = onnxModelRunnerWithThresholds.retrieveOnnxModelRunner(partner);
+            throttlingThresholds = onnxModelRunnerWithThresholds.retrieveThreshold(partner);
+
+            if (onnxModelRunner == null || throttlingThresholds == null) {
+                throw new PreBidException("Cache was empty, fetching and put artefacts for next request");
+            }
+
+            final GreenbidsInferenceData greenbidsInferenceData = new GreenbidsInferenceData(jacksonMapper, database);
+            greenbidsInferenceData.prepareData(bidRequest);
+
+            final FilterService filterService = new FilterService();
+            final Double threshold = partner.getThreshold(throttlingThresholds);
+            impsBiddersFilterMap = filterService.runModeAndFilterBidders(
+                    onnxModelRunner,
+                    greenbidsInferenceData.throttlingMessages,
+                    greenbidsInferenceData.throttlingInferenceRows,
+                    threshold);
+        } catch (PreBidException e) {
+            return Future.succeededFuture(toInvocationResult(
+                    bidRequest, null, InvocationAction.no_action));
+        }
+
+        final GreenbidsInvocationResult greenbidsInvocationResult = new GreenbidsInvocationResult();
+        greenbidsInvocationResult.prepareInvocationResult(partner, bidRequest, impsBiddersFilterMap);
+
+        return Future.succeededFuture(toInvocationResult(
+                greenbidsInvocationResult.updatedBidRequest,
+                greenbidsInvocationResult.analyticsResult,
+                greenbidsInvocationResult.invocationAction));
+    }
+
+    private Partner parseBidRequestExt(BidRequest bidRequest) {
+        return Optional.ofNullable(bidRequest)
+                .map(BidRequest::getExt)
+                .map(ExtRequest::getPrebid)
+                .map(ExtRequestPrebid::getAnalytics)
+                .filter(this::isNotEmptyObjectNode)
+                .map(analytics -> (ObjectNode) analytics.get(BID_REQUEST_ANALYTICS_EXTENSION_NAME))
+                .map(this::toPartner)
+                .orElse(null);
+    }
+
+    private boolean isNotEmptyObjectNode(JsonNode analytics) {
+        return analytics != null && analytics.isObject() && !analytics.isEmpty();
+    }
+
+    private Partner toPartner(ObjectNode adapterNode) {
+        try {
+            return jacksonMapper.mapper().treeToValue(adapterNode, Partner.class);
+        } catch (JsonProcessingException e) {
+            throw new PreBidException("Error decoding bid request analytics extension: " + e.getMessage(), e);
+        }
+    }
+
+    private InvocationResult<AuctionRequestPayload> toInvocationResult(
+            BidRequest bidRequest,
+            AnalyticsResult analyticsResult,
+            InvocationAction action
+    ) {
```

Comment:
according to the code style: closed parenthesis should be on the same line with the instruction it closes

also if you chop down the list of methods parameters you have to add an empty line before the first line of the method

please double-check other places as well (I've seen this code style issue a few times)

Reply:
fixed code style